homeassistant/components/neurio_energy/sensor.py

```diff
@@ -99,35 +172,94 @@ def active_power(self):
         """Return latest active power value."""
         return self._active_power
 
+    @property
+    def active_generation(self):
+        """Return latest active power value."""
+        return self._active_generation
+
+    @property
+    def daily_generation(self):
+        """Return latest active power value."""
+        return self._daily_generation
+
+    @property
+    def net_consumption(self):
+        """Return latest active power value."""
+        return self._net_consumption
+
+    def get_dataset(self):
+        """Get the most recent data from the api once."""
+        start_time = dt_util.start_of_local_day() \
+            .astimezone(dt_util.UTC).isoformat()
+        end_time = dt_util.utcnow().isoformat()
+
+        _LOGGER.debug('Start: %s, End: %s', start_time, end_time)
+
+        try:
+            self._dataset = self.neurio_client \
+                .get_samples_stats(
+                    self.sensor_id, start_time, 'days', end_time)
+        except (requests.exceptions.RequestException, ValueError, KeyError):
+            _LOGGER.warning("Could not update dataset")
+            return None
+
     def get_active_power(self):
         """Return current power value."""
+        header = {'Authorization': 'bearer <token>'}
+        resp = requests.get('http://'+self.sensor_ip+'/current-sample',
```

Comment:
This needs to be added to the `neurio` package. Home Assistant does not contain protocol specific information.

Reply:
Not sure what you mean.

Reply:
Please make a pull request to the neurio library repo and add the new data fetching methods. Once a new version of the library has been released to pypi. A pull request can be made to home assistant which upgrades the used library version and uses the new methods.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#4-communication-with-devices-services

Reply:
I see. I figured my component would just replace the original. I'll make the PR there.

Reply:
Isn't this what I did?

Reply:
I don't think so. Here's the repo of the neurio library:
https://github.com/jordanh/neurio-python

You should submit a PR there to add the necessary new methods. After that is accepted there needs to be a new version released of the neurio library to PyPI. When that is done we can bump the version of the neurio library used in home assistant in another PR.